### PR TITLE
[ResourceList] Fixed loading state z-index

### LINKED ISSUE #3605: ResourceList loading overlay appears under content
## Issue summary
On the [ResourceList](https://polaris.shopify.com/components/lists-and-tables/resource-list) component, the loading spinner 
and overlay don't appear above the resourcelist content. In my (naive) investigation, it looks like the issue is that `SpinnerContainer` has `z-index: 3` and LoadingOverlay has `z-index: 2`, but the `Polaris-ResourceItem__Container` also has `z-index: 2` so there's a conflict.

## Expected behavior
Loading spinner and overlay should appear above all ResourceList content. (note: I've adjusted the overlay color so it's easier to see)
Screen Shot 2020-11-05 at 1 10 15 PM

## Actual behavior
Spinner appears over content but overlay appears underneath.
Screen Shot 2020-11-05 at 1 08 50 PM

## Reduced test case

https://codesandbox.io/s/compassionate-monad-2pxxn?file=/App.js

## Specifications

```bash
  System:
    OS: macOS 10.15.5
    CPU: (12) x64 Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
    Memory: 226.77 MB / 32.00 GB
    Shell: 5.7.1 - /bin/zsh
  Binaries:
    Node: 12.18.0 - ~/.nvm/versions/node/v12.18.0/bin/node
    Yarn: 1.22.4 - ~/.dev/yarn/1.22.4/bin/yarn
    npm: 6.14.4 - ~/.nvm/versions/node/v12.18.0/bin/npm
    Watchman: 4.9.0 - /usr/local/bin/watchman
  Browsers:
    Chrome: 86.0.4240.183
    Chrome Canary: 88.0.4316.0
    Firefox: 82.0.2
    Firefox Developer Edition: 83.0
    Safari: 13.1.1
  npmPackages:
    @shopify/polaris: ^5.10.0 => 5.10.0 
    react: ~16.14.0 => 16.14.0 
    react-dom: ~16.14.0 => 16.14.0 
```

### PR DESCRIPTION
**Pull request by lhoffbeck**
### WHY are these changes introduced?
Fixes #3605 

Bumped two z-index vars.

👁️ **Before** 👁️
(image / copy appear over loading overlay)
before

👁️ **After** 👁️
(image / copy appear under loading overlay)
after

## 

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris-react/blob/master/README.md#development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Versioning%20and%20changelog.md)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';

import {Page, ResourceList, Thumbnail, ResourceItem} from '../src';

export function Playground() {
  return (
    <Page title="Playground">
      <ModalExample />
    </Page>
  );
}

function ModalExample() {
  const resourceName = {
    singular: 'price',
    plural: 'prices',
  };

  return (
    <ResourceList
      resourceName={resourceName}
      items={['one', 'two', 'three', 'four', 'five']}
      renderItem={renderItem}
      loading
    />
  );
}

function renderItem(someText: string) {
  const resourceItemMedia = (
    <Thumbnail
      source="https://burst.shopifycdn.com/photos/black-leather-choker-necklace_373x@2x.jpg"
      alt="Black choker necklace"
    />
  );

  return (
    <ResourceItem id={someText} onClick={() => {}} media={resourceItemMedia}>
      {someText}
    </ResourceItem>
  );
}

```

</details>

### 🎩 checklist

- [x] Tested on [mobile](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md#cross-browser-testing)
- [x] Tested on [multiple browsers](https://help.shopify.com/en/manual/intro-to-shopify/shopify-admin/supported-browsers)
- [x] Tested for [accessibility](https://github.com/Shopify/polaris-react/blob/master/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting%20documentation.md) changes in the style guide
- [ ] For visual design changes, pinged one of @ HYPD, @ mirualves, @ sarahill, or @ ry5n to update the Polaris UI kit

### CODE REVIEW

File: UNRELEASED.md [hunk 1]
```diff
@@ -23,6 +23,7 @@ Use [the changelog guidelines](https://git.io/polaris-changelog-guidelines) to f
 - Fixed `Loading` setting state after it has unmounted ([#4158](https://github.com/Shopify/polaris-react/pull/4158))
 - Prevent extra right margin being added to the `Filter` component when used without filters. ([#4134](https://github.com/Shopify/polaris-react/pull/4134))
 - Fixed offset in `DualThumb` when used with a min value different from 0 [#4172](https://github.com/Shopify/polaris-react/pull/4172)
+- Fixed loading state stacking in `ResourceList` [#4208](https://github.com/Shopify/polaris-react/issues/4208)
```
Comment (kyledurand) [inline UNRELEASED.md:26]:
```suggestion
- Fixed loading state stacking in `ResourceList` ([#4208](https://github.com/Shopify/polaris-react/issues/4208))
```

File: src/components/ResourceList/_variables.scss [hunk 2]
```diff
@@ -8,8 +8,8 @@ $resource-list-variables: (
   bulk-actions-wrapper-stacking-order: 2,
   header-outer-wrapper-stacking-order: 2,
   header-overlay-stacking-order: 4,
-  spinner-stacking-order: 3,
-  overlay-stacking-order: 2,
+  spinner-stacking-order: 4,
+  overlay-stacking-order: 3,
```
Comment (kyledurand) [inline src/components/ResourceList/_variables.scss:12]: Can you move this up to line 10 to maintain the stacking order visually?

**Review by kyledurand — APPROVED** [review]
Changing `overlay-stacking-order` makes me slightly worried but as far as I can tell it should be fine. The only thing I'd recommend is building Polaris for web and 🎩 ing pages that use the Polaris resource list. I think customers list does.

**Comment by lhoffbeck** [comment]
> Changing `overlay-stacking-order` makes me slightly worried
I checked it with a number of different use cases and it was okay, but yeah same 😅  
> The only thing I'd recommend is building Polaris for web and 🎩 ing pages that use the Polaris resource list. I think customers list does.
Good call, will do before merging!

**Comment by lhoffbeck** [comment]
Checked in Web, looks good :)

**Pull request merged**